lms/djangoapps/instructor/enrollment.py

```diff
@@ -267,6 +269,15 @@ def reset_student_attempts(course_id, student, module_state_key, requesting_user
 
     if delete_module:
         module_to_reset.delete()
+        if block and block.has_score:
```

Comment:
I'm thinking this through again. I don't know exactly how to implement it because:

(1) If a block is no longer in the modulestore, we cannot retrieve it from the modulestore.
(2) If we cannot retrieve a block from the modulestore, then we cannot determine `has_score` for the block (right?).
(3) If we cannot determine `has_score` for the missing block, then we should not fire `SCORE_CHANGED` blindly, because we may fire an event for a block that never had a score in the first place.

We cannot guarantee that only problem locations will be input on the instructor dashboard in the "delete student state" interface, and the staff debug toolbar allows deleting student state for any block in the course, anyway.

@nasthagiri @efischer19 thoughts?